### PR TITLE
Dynamic Dashboard: Unit tests for store performance view model

### LINKED ISSUE #12465: Rewrite Store Stats card in SwiftUI following new design
image

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #12465 

## Description

This PR adds tests for the new `StorePerformanceViewModel`

## Testing instructions

Just CI passing is sufficient

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+        // Then
+        XCTAssertEqual(viewModel.siteVisitStatMode, .redactedDueToJetpack)
+    }
+}
+
+// MARK: - Private helpers
+//
+private extension StorePerformanceViewModelTests {
+    func mockSyncAllStats(with stores: MockStoresManager,
```
Comment (hafizrahman) [inline WooCommerce/WooCommerceTests/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModelTests.swift:298]: nice function 👍🏼

**Review by hafizrahman — APPROVED** [review]
This is good to go (after checking the check failures)

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/196">18.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**